README.md

```diff
@@ -61,10 +44,10 @@ LOCAL_MODEL_PATH is required. The two AWS access variables are required unless s
 
 To debug your configuration, try opening a Python shell and run `from spacer import config`, then `config.check()`.
 
-#### Docker build
-The docker build is the preferred build and the one used in deployment.
+### Docker build
+The docker build is used in coralnet's deployment.
 * Install docker on your system
-* Create `secrets.json` as detailed above.
+* Set up configuration as detailed above.
 * Create folder `/path/to/your/local/models` for caching model files.
```

Comment:
Why can't this step be done in the Dockerfile? Oh I see because the pattern is to mount a local dir? In that case maybe a pointer here would help about why you'd want to have access to cached models outside the container?

Reply:
There's a line below this from Oscar, which says this pattern "will make sure the downloaded models are cached to your host storage, which makes rerunning stuff much faster." Running stuff meaning building the Docker image. One such model file is 527 MB (although that's for the old Caffe extractor; the big file for the newer extractor is 66 MB).

Now, what the current Dockerfile *actually* does is redownload the models even if they already exist locally; but the local-dir mount means it *could* do an existence check and avoid redownloading. Also, it does still save on storage space, by not having to store one copy of the models per Docker image (when developing/testing on Docker, it'll be common to have multiple built images stored at a given time).

But this isn't necessary for a production build to work. Even in CoralNet's use cases, the outside-container environment would only be available when testing the Docker image in the environment where it was built, not when the Docker image is running on instances spun up by AWS Batch (through mailman.py).

So perhaps the models-local-dir mount should be an optional step in pyspacer's Dockerfile(s).

The second local-dir mount may also not be necessary if using environment variables instead of secrets.json. Certainly some things to think about, but all this Dockerfile splitting/reworking would be for a separate PR.